crates/figma_import/src/toolkit_schema.rs

```diff
@@ -230,3 +278,140 @@ impl View {
+        let mut values_by_mode: HashMap<String, VariableValue> = HashMap::new();
+        for (mode_id, value) in map.iter() {
+            values_by_mode.insert(mode_id.clone(), VariableValue::from_figma_value(value));
+        }
+        VariableValueMap { values_by_mode }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub enum VariableType {
+    Bool,
+    Number,
+    Text,
+    Color,
+}
+
+// Representation of a Figma variable. We convert a figma_schema::Variable into
+// this format to make the fields a bit easier to access.
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct Variable {
+    pub id: String,
+    pub name: String,
+    pub remote: bool,
+    pub key: String,
+    pub variable_collection_id: String,
+    pub var_type: VariableType,
+    pub values_by_mode: VariableValueMap,
+}
+impl Variable {
+    pub fn from_figma_var(v: &figma_schema::Variable) -> Variable {
+        match v {
+            figma_schema::Variable::Boolean { common, values_by_mode } => Variable {
+                id: common.id.clone(),
+                name: common.name.clone(),
+                remote: common.remote,
+                key: common.key.clone(),
+                variable_collection_id: common.variable_collection_id.clone(),
+                var_type: VariableType::Bool,
+                values_by_mode: VariableValueMap::from_figma_map(values_by_mode),
+                //values_by_mode2,
+            },
+            figma_schema::Variable::Float { common, values_by_mode } => Variable {
+                id: common.id.clone(),
+                name: common.name.clone(),
+                remote: common.remote,
+                key: common.key.clone(),
+                variable_collection_id: common.variable_collection_id.clone(),
+                var_type: VariableType::Number,
+                values_by_mode: VariableValueMap::from_figma_map(values_by_mode),
+            },
+            figma_schema::Variable::String { common, values_by_mode } => Variable {
+                id: common.id.clone(),
+                name: common.name.clone(),
+                remote: common.remote,
+                key: common.key.clone(),
+                variable_collection_id: common.variable_collection_id.clone(),
+                var_type: VariableType::Text,
+                values_by_mode: VariableValueMap::from_figma_map(values_by_mode),
+            },
+            figma_schema::Variable::Color { common, values_by_mode } => Variable {
+                id: common.id.clone(),
+                name: common.name.clone(),
+                remote: common.remote,
+                key: common.key.clone(),
+                variable_collection_id: common.variable_collection_id.clone(),
+                var_type: VariableType::Color,
+                values_by_mode: VariableValueMap::from_figma_map(values_by_mode),
+            },
+        }
```

Comment:
This is also very duplicated, can this also be generic, and can this be combined with the "XorVar" structs?

Reply:
This Variable type contains data about the variable such as its id, collection, and all its values for each mode. It's different than the "XorVar" types, so I think it should be separate, but I will clean this code up to use a common initialization function.